crates/rethnet_evm/benches/state.rs

```diff
@@ -0,0 +1,133 @@
+use criterion::{criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion};
+use revm::{db::StateRef, primitives::AccountInfo};
+use std::{clone::Clone, str::FromStr};
+
+use rethnet_eth::Address;
+use rethnet_evm::state::{HybridState, LayeredState, RethnetLayer, StateError, SyncState};
+
+#[derive(Default)]
+struct RethnetStates {
+    layered: LayeredState<RethnetLayer>,
+    hybrid: HybridState<RethnetLayer>,
+}
+
+impl RethnetStates {
+    fn fill(&mut self, number_of_accounts: u64, number_of_accounts_per_checkpoint: u64) {
+        let mut states: [&mut dyn SyncState<StateError>; 2] = [&mut self.layered, &mut self.hybrid];
+        for state in states.iter_mut() {
+            let number_of_checkpoints = number_of_accounts / number_of_accounts_per_checkpoint;
+            for _ in 0..=number_of_checkpoints {
+                for i in 1..=number_of_accounts {
+                    state
+                        .insert_account(
+                            Address::from_str(&format!("0x{:0>40x}", i)).unwrap(),
+                            AccountInfo::default(),
+                        )
+                        .unwrap();
+                }
+                state.checkpoint().unwrap();
+            }
+        }
+    }
+
+    /// Returns a set of factories, each member of which produces a clone of one of the state objects in this struct.
+    fn make_clone_factories(
+        &self,
+    ) -> Vec<(
+        &'static str, // label of the type of state produced by this factory
+        Box<dyn Fn() -> Box<dyn SyncState<StateError>> + '_>,
+    )> {
+        vec![
+            ("LayeredState", Box::new(|| Box::new(self.layered.clone()))),
+            ("HybridState", Box::new(|| Box::new(self.hybrid.clone()))),
+        ]
+    }
+}
+
+const ADDRESS_SCALES: [u64; 4] = [100, 500, 1000, 2000];
+const CHECKPOINT_SCALES: [u64; 4] = [1, 2, 4, 8];
```

Comment:
```suggestion
const NUM_SCALES: usize = 4;
const CHECKPOINT_SCALES: [u64; NUM_SCALES] = [1, 5, 10, 20];
const MAX_CHECKPOINT_SCALE: u64 = CHECKPOINT_SCALES[NUM_SCALES - 1];
const ADDRESS_SCALES: [u64; 4] =  [
    MAX_CHECKPOINT_SCALE * 5,
    MAX_CHECKPOINT_SCALE * 25,
    MAX_CHECKPOINT_SCALE * 50,
    MAX_CHECKPOINT_SCALE * 100,
];
```

This ensures we can always divide the address scale by the checkpoint scale, avoiding flooring by division.